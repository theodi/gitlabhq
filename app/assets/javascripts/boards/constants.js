export const BoardType = {
  project: 'project',
  group: 'group',
};

export const ListType = {
  assignee: 'assignee',
  milestone: 'milestone',
  backlog: 'backlog',
  closed: 'closed',
  label: 'label',
  promotion: 'promotion',
  blank: 'blank',
};

export const inactiveId = 0;

export const ISSUABLE = 'issuable';
export const LIST = 'list';

export default {
  BoardType,
  ListType,
};
