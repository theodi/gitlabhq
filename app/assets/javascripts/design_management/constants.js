// WARNING: replace this with something
// more sensical as per https://gitlab.com/gitlab-org/gitlab/issues/118611
export const VALID_DESIGN_FILE_MIMETYPE = {
  mimetype: 'image/*',
  regex: /image\/.+/,
};

// https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/types
export const VALID_DATA_TRANSFER_TYPE = 'Files';

export const ACTIVE_DISCUSSION_SOURCE_TYPES = {
  pin: 'pin',
  discussion: 'discussion',
  url: 'url',
};

export const DESIGN_DETAIL_LAYOUT_CLASSLIST = ['design-detail-layout', 'overflow-hidden', 'm-0'];
