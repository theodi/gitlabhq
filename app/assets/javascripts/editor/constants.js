import { __ } from '~/locale';

export const EDITOR_LITE_INSTANCE_ERROR_NO_EL = __(
  '"el" parameter is required for createInstance()',
);

export const URI_PREFIX = 'gitlab';
export const CONTENT_UPDATE_DEBOUNCE = 250;
