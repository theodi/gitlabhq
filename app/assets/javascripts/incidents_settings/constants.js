import { __, s__ } from '~/locale';

/* Integration tabs constants */
export const INTEGRATION_TABS_CONFIG = [
  {
    title: s__('IncidentSettings|Alert integration'),
    component: 'AlertsSettingsForm',
    active: true,
  },
  {
    title: s__('IncidentSettings|PagerDuty integration'),
    component: 'PagerDutySettingsForm',
    active: true,
  },
  {
    title: s__('IncidentSettings|Grafana integration'),
    component: '',
    active: false,
  },
];

export const I18N_INTEGRATION_TABS = {
  headerText: s__('IncidentSettings|Incidents'),
  expandBtnLabel: __('Expand'),
  subHeaderText: s__(
    'IncidentSettings|Set up integrations with external tools to help better manage incidents.',
  ),
};

/* Alerts integration settings constants */

export const I18N_ALERT_SETTINGS_FORM = {
  saveBtnLabel: __('Save changes'),
  introText: __('Action to take when receiving an alert. %{docsLink}'),
  introLinkText: __('More information.'),
  createIssue: {
    label: __('Create an issue. Issues are created for each alert triggered.'),
  },
  issueTemplate: {
    label: __('Issue template (optional)'),
  },
  sendEmail: {
    label: __('Send a separate email notification to Developers.'),
  },
  autoCloseIncidents: {
    label: __('Automatically close incident issues when the associated Prometheus alert resolves.'),
  },
};

export const NO_ISSUE_TEMPLATE_SELECTED = { key: '', name: __('No template selected') };
export const TAKING_INCIDENT_ACTION_DOCS_LINK =
  '/help/operations/metrics/alerts#trigger-actions-from-alerts';
export const ISSUE_TEMPLATES_DOCS_LINK =
  '/help/user/project/description_templates#creating-issue-templates';

/* PagerDuty integration settings constants */

export const I18N_PAGERDUTY_SETTINGS_FORM = {
  introText: s__(
    'PagerDutySettings|Setting up a webhook with PagerDuty will automatically create a GitLab issue for each PagerDuty incident.',
  ),
  activeToggle: {
    label: s__('PagerDutySettings|Active'),
  },
  webhookUrl: {
    label: s__('PagerDutySettings|Webhook URL'),
    helpText: s__(
      'PagerDutySettings|Create a GitLab issue for each PagerDuty incident by %{docsLink}',
    ),
    helpDocsLink: s__('PagerDutySettings|configuring a webhook in PagerDuty'),
    resetWebhookUrl: s__('PagerDutySettings|Reset webhook URL'),
    copyToClipboard: __('Copy'),
    updateErrMsg: s__('PagerDutySettings|Failed to update Webhook URL'),
    updateSuccessMsg: s__('PagerDutySettings|Webhook URL update was successful'),
    restKeyInfo: s__(
      "PagerDutySettings|Resetting the webhook URL for this project will require updating this integration's settings in PagerDuty.",
    ),
  },
  saveBtnLabel: __('Save changes'),
};

export const CONFIGURE_PAGERDUTY_WEBHOOK_DOCS_LINK = 'https://support.pagerduty.com/docs/webhooks';

/* common constants */
export const ERROR_MSG = __('There was an error saving your changes.');
