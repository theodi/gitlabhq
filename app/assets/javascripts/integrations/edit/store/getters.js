export const isInheriting = state => (state.defaultState === null ? false : !state.override);

export const isSavingOrTesting = state => state.isSaving || state.isTesting;

export const propsSource = (state, getters) =>
  getters.isInheriting ? state.defaultState : state.customState;

export const currentKey = (state, getters) => (getters.isInheriting ? 'admin' : 'custom');
