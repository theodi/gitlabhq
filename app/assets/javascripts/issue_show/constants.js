import { __ } from '~/locale';

export const IssuableStatus = {
  Open: 'opened',
  Closed: 'closed',
};

export const IssuableStatusText = {
  [IssuableStatus.Open]: __('Open'),
  [IssuableStatus.Closed]: __('Closed'),
};

export const IssuableType = {
  Issue: 'issue',
  Epic: 'epic',
  MergeRequest: 'merge_request',
};

export const STATUS_PAGE_PUBLISHED = __('Published on status page');
export const JOIN_ZOOM_MEETING = __('Join Zoom meeting');
