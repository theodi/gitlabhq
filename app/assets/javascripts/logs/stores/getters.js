import { formatDate } from '../utils';

const mapTrace = ({ timestamp = null, pod = '', message = '' }) =>
  [timestamp ? formatDate(timestamp) : '', pod, message].join(' | ');

export const trace = state => state.logs.lines.map(mapTrace).join('\n');

export const showAdvancedFilters = state => {
  if (state.environments.current) {
    const environment = state.environments.options.find(
      ({ name }) => name === state.environments.current,
    );

    return Boolean(environment?.enable_advanced_logs_querying);
  }
  const managedApp = state.managedApps.options.find(
    ({ name }) => name === state.managedApps.current,
  );

  return Boolean(managedApp?.enable_advanced_logs_querying);
};
