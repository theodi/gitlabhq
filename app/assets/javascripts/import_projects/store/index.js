import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import actionsFactory from './actions';
import * as getters from './getters';
import mutations from './mutations';

Vue.use(Vuex);

export default ({ initialState, endpoints, hasPagination }) =>
  new Vuex.Store({
    state: { ...state(), ...initialState },
    actions: actionsFactory({ endpoints, hasPagination }),
    mutations,
    getters,
  });
