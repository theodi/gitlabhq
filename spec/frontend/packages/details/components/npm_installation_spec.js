import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { npmPackage as packageEntity } from 'jest/packages/mock_data';
import { registryUrl as nugetPath } from 'jest/packages/details/mock_data';
import NpmInstallation from '~/packages/details/components/npm_installation.vue';
import CodeInstructions from '~/vue_shared/components/registry/code_instruction.vue';
import { TrackingActions } from '~/packages/details/constants';
import { npmInstallationCommand, npmSetupCommand } from '~/packages/details/store/getters';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('NpmInstallation', () => {
  let wrapper;

  const findCodeInstructions = () => wrapper.findAll(CodeInstructions);

  function createComponent() {
    const store = new Vuex.Store({
      state: {
        packageEntity,
        nugetPath,
      },
      getters: {
        npmInstallationCommand,
        npmSetupCommand,
      },
    });

    wrapper = shallowMount(NpmInstallation, {
      localVue,
      store,
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    if (wrapper) wrapper.destroy();
  });

  it('renders all the messages', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('installation commands', () => {
    it('renders the correct npm command', () => {
      expect(
        findCodeInstructions()
          .at(0)
          .props(),
      ).toMatchObject({
        instruction: 'npm i @Test/package',
        multiline: false,
        trackingAction: TrackingActions.COPY_NPM_INSTALL_COMMAND,
      });
    });

    it('renders the correct yarn command', () => {
      expect(
        findCodeInstructions()
          .at(1)
          .props(),
      ).toMatchObject({
        instruction: 'yarn add @Test/package',
        multiline: false,
        trackingAction: TrackingActions.COPY_YARN_INSTALL_COMMAND,
      });
    });
  });

  describe('setup commands', () => {
    it('renders the correct npm command', () => {
      expect(
        findCodeInstructions()
          .at(2)
          .props(),
      ).toMatchObject({
        instruction: 'echo @Test:registry=undefined/ >> .npmrc',
        multiline: false,
        trackingAction: TrackingActions.COPY_NPM_SETUP_COMMAND,
      });
    });

    it('renders the correct yarn command', () => {
      expect(
        findCodeInstructions()
          .at(3)
          .props(),
      ).toMatchObject({
        instruction: 'echo \\"@Test:registry\\" \\"undefined/\\" >> .yarnrc',
        multiline: false,
        trackingAction: TrackingActions.COPY_YARN_SETUP_COMMAND,
      });
    });
  });
});
