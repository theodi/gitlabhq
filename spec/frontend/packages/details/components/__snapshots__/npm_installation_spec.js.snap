// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NpmInstallation renders all the messages 1`] = `
<div>
  <h3
    class="gl-font-lg"
  >
    Installation
  </h3>
   
  <code-instruction-stub
    copytext="Copy npm command"
    instruction="npm i @Test/package"
    label="npm command"
    trackingaction="copy_npm_install_command"
    trackinglabel="code_instruction"
  />
   
  <code-instruction-stub
    copytext="Copy yarn command"
    instruction="yarn add @Test/package"
    label="yarn command"
    trackingaction="copy_yarn_install_command"
    trackinglabel="code_instruction"
  />
   
  <h3
    class="gl-font-lg"
  >
    Registry setup
  </h3>
   
  <code-instruction-stub
    copytext="Copy npm setup command"
    instruction="echo @Test:registry=undefined/ >> .npmrc"
    label="npm command"
    trackingaction="copy_npm_setup_command"
    trackinglabel="code_instruction"
  />
   
  <code-instruction-stub
    copytext="Copy yarn setup command"
    instruction="echo \\\\\\"@Test:registry\\\\\\" \\\\\\"undefined/\\\\\\" >> .yarnrc"
    label="yarn command"
    trackingaction="copy_yarn_setup_command"
    trackinglabel="code_instruction"
  />
   
  <gl-sprintf-stub
    message="You may also need to setup authentication using an auth token. %{linkStart}See the documentation%{linkEnd} to find out more."
  />
</div>
`;
